android/guava-tests/test/com/google/common/collect/ListsTest.java

```diff
@@ -81,7 +81,7 @@ public class ListsTest extends TestCase {
   private static final class RemoveFirstFunction implements Function<String, String>, Serializable {
     @Override
     public String apply(String from) {
-      return (from.length() == 0) ? from : from.substring(1);
+      return (from.isEmpty()) ? from : from.substring(1);
```

Comment:
The wrapping parens can now be dropped as well.

Reply:
yes, this is a new recipe then, as the goal is not to touch this, just leveraging rewrite for doing all this kind of trivial stuff. Thats exactly what its about. Thanks for demonstrating why its a good invest to spend some CPU time avoiding this upfront. No offence given, really appreciate your feedback and guidance. This is just the perfect example what this is all about.

@cpovirk
@romani

- https://docs.openrewrite.org/recipes/staticanalysis/unnecessaryparentheses

Reply:
> can now be dropped as well

Assuming its just as obsolete, as it was before, indicating the broken window effect bleeding the code, due to absents of (proper) quality gate.